server/src/main/java/com/cloud/storage/snapshot/SnapshotManager.java

```diff
@@ -58,6 +54,12 @@ public interface SnapshotManager extends Configurable {
 
     public static final ConfigKey<Boolean> VmStorageSnapshotKvm = new ConfigKey<>(Boolean.class, "kvm.vmstoragesnapshot.enabled", "Snapshots", "false", "For live snapshot of virtual machine instance on KVM hypervisor without memory. Requieres qemu version 1.6+ (on NFS or Local file system) and qemu-guest-agent installed on guest VM", true, ConfigKey.Scope.Global, null);
 
+    ConfigKey<Boolean> kvmIncrementalSnapshot = new ConfigKey<>(Boolean.class, "kvm.incremental.snapshot", "Snapshots", "false", "Whether differential snapshots are enabled for" +
+            " KVM or not. When this is enabled, all KVM snapshots will be incremental. Bear in mind that it will generate a new full snapshot when the snapshot chain reaches the limit defined in snapshot.delta.max.", true, ConfigKey.Scope.Cluster, null);
```

Comment:
seems this is dynamic, will this impact in-progress snapshots in certain state/stage?

Reply:
I've tested changing these while taking snapshots, I have not found any issue.

Reply:
what about the recurring snapshots in progress when taking/backing up snapshots? any chance to have some snapshots with incremental and some with non-incremental?

Reply:
@sureshanaparti yes this can happen, and was tested by @winterhazel (see https://github.com/apache/cloudstack/pull/9270#pullrequestreview-2242776880) there shouldn't be any issue with some incremental and some non-incremental snapshots for the same volume.